### PR TITLE
Cleanup tf2_tools to be more modern.

### PR DESCRIPTION
**Pull request by clalancette**
In short:
1.  Change it from an ament_cmake -> ament_python project
2.  Fix up the dependencies to be more correct
3.  Make it a console script

Note that this likely still only works on Linux due to its
dependency on a subprocess call to dot, but at least it is
more modern.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

### CODE REVIEW

File: tf2_tools/setup.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    packages=[package_name],
+    data_files=[
+        ('share/ament_index/resource_index/packages',
+            ['resource/' + package_name]),
+        ('share/' + package_name, ['package.xml']),
+    ],
+    install_requires=['setuptools'],
+    zip_safe=True,
+    author='Wim Meeussen',
+    maintainer='Chris Lalancette',
```
Comment (ahcorde) [inline tf2_tools/setup.py:17]: Add me here as a maintainer too ?

**Review by ahcorde — APPROVED** [review]
LGTM

File: tf2_tools/setup.py (hunk 1 above)
Comment (clalancette) [reply tf2_tools/setup.py:17]: As far as I understand, there is no support for multiple maintainers in the setup.py format: https://stackoverflow.com/questions/9999829/how-to-specify-multiple-authors-emails-in-setup-py .

I guess I could add in a comma-separated string of maintainers/email addresses, but I don't think it is totally necessary.  Both of our names are in the package.xml, which is good enough for me.

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13227/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8157/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10942/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/13257/)

**Comment by Timple** [comment]
I think the extension for `view_frames.py` can be dropped right?

The language is an implementation detail not relevant for the end-user. And common practice is to leave it out for nodes.

**Comment by clalancette** [comment]
I don't have a strong preference, honestly.  There is some benefit to leaving it as-is, so we don't break scripts that are already using this.  Nevertheless, if you feel strongly about it, feel free to open a PR and we can discuss it there.

**Pull request merged**